### PR TITLE
Feature/crpl 2862 simplified endpoint error handler

### PR DESCRIPTION
**Pull request by approximata**
Fix server start node by default start with ipv4 port. Possible start with ipv6 too.
Fix error regarding server problem log out from endpoint. Process wont start.

### CODE REVIEW

**Comment by danstocker** [comment]
Need to organize this better
Screenshot 2022-10-27 at 11 40 50

File: io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json [hunk 1]
```diff
@@ -100,8 +100,5 @@
     }
   },
   "iconId": "802b6352-2279-4681-84b8-6558e185fd36",
-  "attributes": {
-    "internal": true,
-    "locked": true
-  }
+  "attributes": {}
```
Comment (csabee1979) [inline io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json:103]: Should be locked

File: io/http/server/endpoint listener/3e022fe6-001b-45a9-9d81-14f1581bef8c.json [hunk 2]
```diff
@@ -350,7 +332,5 @@
     "api": null
   },
   "iconId": "47baa0d3-adcc-4071-8ab3-768676771342",
-  "attributes": {
-    "locked": true
-  }
+  "attributes": {}
```
Comment (csabee1979) [inline io/http/server/endpoint listener/3e022fe6-001b-45a9-9d81-14f1581bef8c.json:335]: Should be locke

File: io/http/server/endpoint listener/group/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json [hunk 3]
```diff
@@ -0,0 +1,107 @@
+{
+  "name": "io/http/server/Endpoint listener/Group",
```
[the remark this answers is not in the document]
Comment (lvass74) [reply io/http/server/endpoint listener/group/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json]: If will be reusable, port names, types and description should be revised.

File: sdks/expressjs/listener/571c94c6-38b1-4212-833a-b947f50fadec.json [hunk 4]
```diff
@@ -8,8 +8,8 @@
         "description": "The id of the express instance.\n\nExample: \n\"my-express-server\""
       },
       "11745554-d9d0-41bb-af77-987d8450ccfe": {
-        "type": "number",
-        "name": "port",
+        "type": "(number or [number, string])",
+        "name": "port & opt host",
```
[the remark this answers is not in the document]
Comment (lvass74) [reply sdks/expressjs/listener/571c94c6-38b1-4212-833a-b947f50fadec.json:12]: +description should be completed for the 2nd use case

File: io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json [hunk 5]
```diff
@@ -0,0 +1,106 @@
+{
+  "name": "io/http/server/Endpoint listener/server error checker",
```
Comment (csabee1979) [inline io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json:2]: Please start the prototype name uppercase: Server error checker

File: io/http/server/endpoint listener/set params/e035b5aa-c5e9-4d32-9623-a42d6a2b1581.json [hunk 6]
```diff
@@ -0,0 +1,82 @@
+{
+  "name": "io/http/server/Endpoint listener/set params",
```
Comment (csabee1979) [inline io/http/server/endpoint listener/set params/e035b5aa-c5e9-4d32-9623-a42d6a2b1581.json:2]: Start the prototype name uppercase: Set params

File: io/http/server/endpoint listener/set resp. content type type/bf64fdcb-0480-4f63-a39f-b8ad7dc1976a.json [hunk 7]
```diff
@@ -0,0 +1,101 @@
+{
+  "name": "io/http/server/Endpoint listener/set resp. content type type",
```
Comment (csabee1979) [inline io/http/server/endpoint listener/set resp. content type type/bf64fdcb-0480-4f63-a39f-b8ad7dc1976a.json:2]: Start the prototype name uppercase: Set resp. content type type

File: io/http/server/endpoint listener/set resp. content type/bf64fdcb-0480-4f63-a39f-b8ad7dc1976a.json [hunk 8]
```diff
@@ -0,0 +1,101 @@
+{
+  "name": "io/http/server/Endpoint listener/set resp. content type",
```
Comment (csabee1979) [inline io/http/server/endpoint listener/set resp. content type/bf64fdcb-0480-4f63-a39f-b8ad7dc1976a.json:2]: Start the prototype name uppercase: Set resp. content type

File: io/http/server/endpoint listener/set resp. content type type/bf64fdcb-0480-4f63-a39f-b8ad7dc1976a.json (hunk 7 above)
Comment (csabee1979) [reply io/http/server/endpoint listener/set resp. content type type/bf64fdcb-0480-4f63-a39f-b8ad7dc1976a.json:2]: Is the doubled type typo? Was  Set resp. content type the intentio?

File: string/url formatter/8f7f2e0d-2a80-428c-904b-c8d603fab102.json [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+  },
+  "implementations": {
+    "cranq": {
+      "nodes": {
+        "2a1aca15-e0fa-4fd4-b8fa-f091b1990f2a": {
+          "coordinates": [
+            266.3333333333333,
+            -182.33333333333337
+          ],
+          "name": "template filler",
```
Comment (csabee1979) [inline string/url formatter/8f7f2e0d-2a80-428c-904b-c8d603fab102.json:27]: Add  more descriptive name.

File: string/url formatter/8f7f2e0d-2a80-428c-904b-c8d603fab102.json [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+          "params": {
+            "545a5e1a-9807-4456-b557-dfa8aa8ef28a": "{\n\"protocol\":\"http\",\n\"port\": \"\",\n\"path\": \"\",\n\"query\": \"\",\n\"hash\": \"\"\n}"
+          }
+        },
+        "7131d44d-c93c-4f08-89ca-d75c781851de": {
+          "coordinates": [
+            -234.66666666666669,
+            -83.33333333333337
+          ],
+          "name": "splitter",
```
Comment (csabee1979) [inline string/url formatter/8f7f2e0d-2a80-428c-904b-c8d603fab102.json:49]: Add  more descriptive name.

File: string/url formatter/8f7f2e0d-2a80-428c-904b-c8d603fab102.json [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+          ],
+          "name": "wrap hostnam if ipv6",
+          "prototype": "3de24890-3a09-4df0-893c-75c34d68af1c"
+        },
+        "f7e2d933-978a-4bd7-b7c7-7d9a03c64c6d": {
+          "coordinates": [
+            160.33333333333331,
+            -5.333333333333371
+          ],
+          "name": "syncer",
```
Comment (csabee1979) [inline string/url formatter/8f7f2e0d-2a80-428c-904b-c8d603fab102.json:68]: Add  more descriptive name.

File: string/url formatter/8f7f2e0d-2a80-428c-904b-c8d603fab102.json [hunk 12]
```diff
@@ -0,0 +1,200 @@
+{
+  "name": "string/url formatter",
+  "description": "format records to url. \n",
+  "interface": {
+    "inputs": {
+      "7e1c79f2-3603-41bc-bfdb-e4e64d27b0a2": {
+        "type": "{\noptional \"protocol\": (\"http\" or \"https\" or \"ws\" or \"wss\"), \n\"hostname\": string,\noptional \"port\": number,\noptional \"path\": string,\noptional \"query\": string,\noptional \"hash\": string\n}",
+        "name": "params",
+        "description": "[Inherited from port `params` of `template filler`] \nThe actual parameters used for filling the template"
```
Comment (csabee1979) [inline string/url formatter/8f7f2e0d-2a80-428c-904b-c8d603fab102.json:9]: Update the descritption to fit the recent port.

File: sdks/expressjs/listener/571c94c6-38b1-4212-833a-b947f50fadec.json [hunk 13]
```diff
@@ -124,21 +146,10 @@
...
-          }
-        },
         "a2965b53-ca20-48fa-bca4-e0a4cc8b1422": {
           "coordinates": [
-            -185.2,
-            -0.19999999999998863
+            -483.2,
+            39.80000000000001
           ],
           "name": "forwarder/double",
```
Comment (csabee1979) [inline sdks/expressjs/listener/571c94c6-38b1-4212-833a-b947f50fadec.json:154]: Add more descriptive instance name.

Comment (csabee1979) [inline sdks/expressjs/listener/571c94c6-38b1-4212-833a-b947f50fadec.json:154]: Add more descriptive instance name.

File: io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json [hunk 14]
```diff
@@ -0,0 +1,106 @@
+{
+  "name": "io/http/server/Endpoint listener/Server error checker",
+  "interface": {
+    "inputs": {
+      "8978272a-83da-48fd-b253-ca7585916edc": {
+        "type": "typeof `unsplit` of `splitter`",
+        "name": "unsplit",
```
Comment (danstocker) [inline io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json:7]: Please name this something meaningful, eg. "endpoint error"

File: io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+        "description": "[Inherited from port `unsplit` of `splitter`] \nReceives records or tuples to be split into individual items.\n\nExamples:\n* {\"a\": 5, \"b\": 2}\n* [5, 2]"
+      }
+    },
+    "outputs": {
+      "e7b7f2c0-4e3f-4630-83d6-f5c18f170947": {
+        "type": "typeof `on true` of `fork`",
+        "name": "on true",
```
Comment (danstocker) [inline io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json:14]: Please name this something meaningful, eg. "server error"

File: io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json [hunk 16]
```diff
@@ -0,0 +1,106 @@
...
+        "description": "[Inherited from port `on true` of `fork`] \nSends signal received via `data` when condition was true."
+      },
+      "bfb24c70-24af-457b-bc18-d945b3f65b3a": {
+        "type": "typeof `on false` of `fork`",
+        "name": "on false",
```
Comment (danstocker) [inline io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json:19]: Please name this something meaningful, eg. "request error"

File: io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json [hunk 17]
```diff
@@ -0,0 +1,106 @@
...
+          "params": {
+            "7b39d568-3d5a-46fe-8670-3355f60c2746": "[\"error\", \"origin\"]"
+          }
+        },
+        "d365c1e4-9785-4817-8313-dc81c63a2240": {
+          "coordinates": [
+            45,
+            106
+          ],
+          "name": "is application error",
```
Comment (danstocker) [inline io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json:43]: What is "application" error? Server error?

File: io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json [hunk 18]
```diff
@@ -0,0 +1,106 @@
+{
+  "name": "io/http/server/Endpoint listener/Server error checker",
+  "interface": {
+    "inputs": {
```
Comment (danstocker) [inline io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json:4]: PLease clean up port types as well

File: string/url formatter/8f7f2e0d-2a80-428c-904b-c8d603fab102.json [hunk 19]
```diff
@@ -0,0 +1,200 @@
+{
+  "name": "string/Url formatter",
```
Comment (danstocker) [inline string/url formatter/8f7f2e0d-2a80-428c-904b-c8d603fab102.json:2]: "URL" should be all caps

File: string/url formatter/8f7f2e0d-2a80-428c-904b-c8d603fab102.json [hunk 20]
```diff
@@ -0,0 +1,200 @@
...
+      "7e1c79f2-3603-41bc-bfdb-e4e64d27b0a2": {
+        "type": "{\noptional \"protocol\": (\"http\" or \"https\" or \"ws\" or \"wss\"), \n\"hostname\": string,\noptional \"port\": number,\noptional \"path\": string,\noptional \"query\": string,\noptional \"hash\": string\n}",
+        "name": "params",
+        "description": "The url parameters used for create an url string."
+      }
+    },
+    "outputs": {
+      "b07917f3-ec1b-484d-96c2-abb9dfb5b041": {
+        "type": "string",
+        "name": "url"
```
Comment (danstocker) [inline string/url formatter/8f7f2e0d-2a80-428c-904b-c8d603fab102.json:15]: All caps

File: io/http/server/simplified endpoint listener/cf29d613-2fc6-4230-8c7f-bd86cb1c4c9e.json [hunk 21]
```diff
@@ -55,13 +55,51 @@
...
+          ],
+          "name": "format url",
+          "prototype": "8f7f2e0d-2a80-428c-904b-c8d603fab102"
+        },
+        "4b2463ea-fab7-4a1e-9ec9-04ef9ae2016b": {
+          "coordinates": [
+            334.5,
+            -315
+          ],
+          "name": "log url",
```
Comment (danstocker) [inline io/http/server/simplified endpoint listener/cf29d613-2fc6-4230-8c7f-bd86cb1c4c9e.json:91]: This is not logging an URL, but the fact that the server started. The name should reflect that

**Review by danstocker — CHANGES_REQUESTED** [review]

File: io/http/server/endpoint listener/group/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (approximata) [reply io/http/server/endpoint listener/group/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json]: this one is deleted.

File: io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json (hunk 17 above)
Comment (approximata) [reply io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json]: rewrote to server.

File: io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json (hunk 18 above)
Comment (approximata) [reply io/http/server/endpoint listener/server error checker/0faadab5-12bc-4bd8-b7e1-0bd8ddf21a43.json:4]: done

File: string/url formatter/8f7f2e0d-2a80-428c-904b-c8d603fab102.json (hunk 19 above)
Comment (approximata) [reply string/url formatter/8f7f2e0d-2a80-428c-904b-c8d603fab102.json]: renamed

**Review by danstocker — APPROVED** [review]

**Review by lvass74 — APPROVED** [review]

**Pull request merged**